### PR TITLE
Mef/zext phi cleanup

### PR DESCRIPTION
**Pull request by cqc-melf**
# Description

I have tried to split this PR into useful commits to make it easier to review:
[regenerate QIR test files](https://github.com/CQCL/pytket-qir/commit/3cf401538ea863ace5284dd2e3cfc6de83815177)
Contains only the regeneration of the QIR files, this is quiet a lot and not really relevant to review.

[clean up tests](https://github.com/CQCL/pytket-qir/commit/c9ca971b0cc186038a89f508f04fd7d1fcf16e93)
In this commit, I have cleaned up the tests and reorganised profile and non profile tests together in on file.

[fix bug in phi node generation, add set_bit via zext](https://github.com/CQCL/pytket-qir/commit/64cee2a0c730f96ccd877de824cfa743c81ae5b0)
This is the most relevant to review, this fixes a problem with the phi nodes and add the new set_bit via zext.

[update version, changelog, pytket](https://github.com/CQCL/pytket-qir/commit/8df7eda06f0358befef51539b86932df662cfb1e)
Contains general cleanup and preparation for the next release candidate. 

# Related issues

Please mention any github issues addressed by this PR.

# Checklist

- [x] I have performed a self-review of my code.
- [x] I have commented hard-to-understand parts of my code.
- [x] I have made corresponding changes to the public API documentation.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have updated the changelog with any user-facing changes.

### CODE REVIEW

File: pytket/qir/conversion/pconversion.py [hunk 1]
```diff
@@ -325,7 +362,9 @@ def get_ssa_list(self, reg_name: str) -> list:
             raise ValueError(f"{reg_name} is not a valid register")
         return self.ssa_vars[reg_name]
 
-    def set_ssa_vars(self, reg_name: str, ssa_i64: Value, trunc: bool = True) -> None:
+    def set_ssa_vars(self, reg_name: str, ssa_i64: Value, trunc: bool = False) -> None:
+        # todo set default value for trunc to true, when enough
```
Comment (cqc-melf) [inline pytket/qir/conversion/pconversion.py:378]: see https://github.com/CQCL/pytket-qir/issues/168

File: docs/changelog.rst [hunk 2]
```diff
@@ -1,11 +1,12 @@
 Changelog
 ~~~~~~~~~
 
-0.13.0rc1 (August 2024)
+0.13.0rc2 (August 2024)
 -----------------------
 
 * Updated pyqir version requirement to 0.10.4.
 * Add option to generate profile compatible QIR
+* Updated pytket version requirement to 1.33.
```
Comment (cqc-alec) [inline docs/changelog.rst]: This is not a stable release; do we really want to require a release candidate?

File: setup.py [hunk 3]
```diff
@@ -45,7 +45,7 @@
     packages=find_namespace_packages(include=["pytket.*"]),
     include_package_data=True,
     install_requires=[
-        "pytket >= 1.31.0",
+        "pytket >= 1.33.0rc0",
```
Comment (cqc-alec) [inline setup.py]: See above comment: shouldn't we just upgrade to 1.32.0?

File: tests/pconversion_test.py
Comment (cqc-alec) [inline tests/pconversion_test.py:1]: Have all these tests moved elsewhere?

File: tests/pwasm_test.py
Comment (cqc-alec) [inline tests/pwasm_test.py:1]: Have all these tests moved elsewhere?

File: tests/pconditional_test.py
Comment (cqc-alec) [inline tests/pconditional_test.py:1]: Have all these tests moved elsewhere?

File: pytket/qir/conversion/pconversion.py [hunk 4]
```diff
@@ -659,7 +700,7 @@ def conv_conditional(self, command: Command, op: Conditional) -> None:
                             ssa_list[-1][1].name != ssa_list[i][1].name
                             and ssa_list[i][1].name[0:5] != "condb"
                         ):
-                            phi.add_incoming(ssa_list[i][0], ssa_list[i][1])
+                            phi.add_incoming(ssa_list[i][0], self.active_block_list[-3])
```
Comment (cqc-alec) [inline pytket/qir/conversion/pconversion.py]: Can you add a comment explaining why "-3"?

File: pytket/qir/conversion/pconversion.py [hunk 5]
```diff
@@ -286,6 +288,8 @@ def _set_bit_in_creg(self, creg: str, index: int, ssa_bit: Value) -> None:
             f"{self.active_block_main.name}_{self.block_count_sb}",
             entry_point,
         )
+        if self.active_block_main.name[0:5] != "condb":
```
Comment (cqc-alec) [inline pytket/qir/conversion/pconversion.py]: Do we need to hard-code this label? Is it guaranteed?

File: tests/conditional_test.py [hunk 6]
```diff
@@ -28,7 +30,14 @@
 from pytket.qir.conversion.api import QIRFormat, pytket_to_qir
 
 
-def test_pytket_qir_conditional() -> None:
+@pytest.mark.parametrize(
```
Comment (cqc-melf) [inline tests/conditional_test.py:33]: This file now contains the tests from `tests/conditional_test.py` and `tests/conditional_test.py` with parametrised tests. I Most of the tests have been in both files anyway, I have added the tests that were in only on of the files. I had to rename a few of the tests to make sure I can have all of the in one file.

File: tests/pconversion_test.py
Comment (cqc-melf) [reply tests/pconversion_test.py:1]: Yes, see comment below. https://github.com/CQCL/pytket-qir/pull/167#discussion_r1766526050

File: tests/pwasm_test.py
Comment (cqc-melf) [reply tests/pwasm_test.py:1]: Yes, see comment below. https://github.com/CQCL/pytket-qir/pull/167#discussion_r1766526050

File: tests/pconditional_test.py
Comment (cqc-melf) [reply tests/pconditional_test.py:1]: Yes, see comment below. https://github.com/CQCL/pytket-qir/pull/167#discussion_r1766526050

File: setup.py (hunk 3 above)
Comment (cqc-melf) [reply setup.py]: done in [bf54ccd](https://github.com/CQCL/pytket-qir/pull/167/commits/bf54ccd1a7f5d80826f1b94b5136efa5e68ec0cd)

File: docs/changelog.rst (hunk 2 above)
Comment (cqc-melf) [reply docs/changelog.rst]: done in [bf54ccd](https://github.com/CQCL/pytket-qir/pull/167/commits/bf54ccd1a7f5d80826f1b94b5136efa5e68ec0cd)

File: pytket/qir/conversion/pconversion.py (hunk 5 above)
Comment (cqc-melf) [reply pytket/qir/conversion/pconversion.py]: I have moved the hardcoded block names to the constructor of the class, so it is easier to change if that would be required.
See [6f4bed0](https://github.com/CQCL/pytket-qir/pull/167/commits/6f4bed04d56fd467109cd6d1c8ba092f0809877b)

File: pytket/qir/conversion/pconversion.py (hunk 4 above)
Comment (cqc-melf) [reply pytket/qir/conversion/pconversion.py]: I have added a few comments to make it easier to understand, if there are still questions / it is not clear, I will add more details on the structure of the active block list.
See [6f4bed0](https://github.com/CQCL/pytket-qir/pull/167/commits/6f4bed04d56fd467109cd6d1c8ba092f0809877b)

File: pytket/qir/conversion/pconversion.py [hunk 7]
```diff
@@ -593,6 +605,40 @@ def _get_ssa_from_cl_bit_op(
         else:
             raise ValueError(f"unsupported bitwise operation {type(bit)}")
 
+    def _add_phi(self) -> None:
+        """
+        add phi nodes for the previously changed registers.
+        phi requires ssa variables from both predecessor blocks,
+        this are not necaserryly the blocks where the variables
```
Comment (cqc-alec) [inline pytket/qir/conversion/pconversion.py]:
```suggestion
        these are not necessarily the blocks where the variables
```

**Review by cqc-alec — DISMISSED** [review]

**Review by cqc-alec — APPROVED** [review]

**Pull request merged**